### PR TITLE
README.md: adds instructions for deploying knative-*

### LINKED ISSUE #27: Update README at all levels
We are missing useful information in the READMEs.

* In the root directory - Information about Knative, how to deploy `operator` alongisde `eventing`, `serving`, or both, as well as examples for an integrated solution.
* At charm/knative-* level - Information about each component as well as how to deploy examples for individual use cases.

### PR DESCRIPTION
**Pull request by DnPlas**
Adds instructions in the root directory to deploy knative-operators. This README also includes a guide to deploy an integration example.

Partially fixes #27

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,3 +3,104 @@
...
+
+```
+juju deploy istio-pilot --config default-gateway=${DEFAULT_GATEWAY} --trust
+juju deploy istio-gateway istio-ingressgateway --config kind="ingress" --trust
+juju relate istio-pilot istio-ingressgateway
+```
+
+### `knative-operator`
+
+The `knative-operator` is responsible for installing, configuring, and managing the `serving` and `eventing` components. Despite your final application, you must deploy the charmed `knative-operator` before any other Knative component.
```
Comment (ca-scribner) [inline README.md:34]: I think you're looking for a different word than "despite".  That kinda means "it doesn't affect your final application, but you must ...".  I don't think you need that part of the phrase anyway, but if you still want it you could say "Regardless of your final application, ..." or "In spite of ..."

```suggestion
The `knative-operator` is responsible for installing, configuring, and managing the `serving` and `eventing` components. You must deploy the charmed `knative-operator` before any other Knative component.
```

File: README.md [hunk 2]
```diff
@@ -3,3 +3,104 @@
...
+juju deploy knative-operator --trust
+```
+
+### `knative-serving`
+
+`knative-serving` provides components that enable rapid deployment of serverless containers, autoscaling (including down to zero), support for multiple layers, and revision tracking. Charmed `knative-serving` can be deployed as follows:
+
+```
+juju deploy knative-serving --config namespace="knative-serving" --config istio.gateway.namespace=${MODEL_NAME} --trust
+```
```
Comment (ca-scribner) [inline README.md:46]: Can you add explanations for these?  Something like below.  Also, I'd put the `istio.gateway.name`, even though it happens to work with the default.  Good chance someone deploying this wont actually use the `knative-gateway` name applied above, which then means they need to know about this config.


Sorry, gh doesn't like "suggestions" that include backticks.  Below is a suggested update to the highlighted section:

```
juju deploy knative-serving --config namespace="knative-serving" --config istio.gateway.namespace=${MODEL_NAME} --trust
```

where:
* `namespace`: The namespace `knative-serving` will be deployed into (it cannot be deployed into the same namespace as `knative-operator` or `knative-eventing`
* istio.gateway.namespace: The namespace the Istio gateway is deployed to (generally, the model that Istio is deployed to)
* istio.gateway.name: The name of the Istio gateway

File: README.md [hunk 3]
```diff
@@ -3,3 +3,104 @@
...
+
+### `knative-eventing`
+
+`knative-eventing` provides tools for routing events from producers to sinks, enabling developers to use an event-driven architecture with their applications. Charmed `knative-eventing` can be deployed as follows:
+
+```
+juju deploy knative-eventing --config namespace="knative-eventing" --trust
+```
```
Comment (ca-scribner) [inline README.md:54]: same as above

File: README.md [hunk 4]
```diff
@@ -3,3 +3,104 @@
...
+
+## Integration example
+
+To run an simple example that integrates both `eventing` and `serving`, you can run the one provided in `examples/`.
```
Comment (ca-scribner) [inline README.md:58]:
```suggestion
To run a simple example that integrates both `eventing` and `serving`, you can run the one provided in `examples/`.
```

**Review by ca-scribner — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (DnPlas) [reply README.md:46]: I just changed the `namespace` definition a bit, we have to specify it's the resources not the charm this namespace has effect on.

**Pull request merged**